Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Components/Dev/TokenUrlComponent.cs

```diff
@@ -0,0 +1,133 @@
+public class TokenUrlComponent : GH_Component
+{
+  public TokenUrlComponent()
+    : base(
+      "Speckle Model URL",
+      "URL",
+      "Create a Speckle model link using URL and developer token",
+      ComponentCategories.PRIMARY_RIBBON,
+      ComponentCategories.DEVELOPER
+    ) { }
+
+  public override Guid ComponentGuid => GetType().GUID;
+  protected override Bitmap Icon => Resources.speckle_inputs_model;
+
+  protected override void RegisterInputParams(GH_InputParamManager pManager)
+  {
+    pManager.AddTextParameter("Speckle Url", "Url", "Speckle URL", GH_ParamAccess.item);
+    pManager.AddTextParameter("Speckle Token", "Token", "Speckle Authorization Token", GH_ParamAccess.item);
+  }
+
+  protected override void RegisterOutputParams(GH_OutputParamManager pManager)
+  {
+    pManager.AddParameter(new SpeckleUrlModelResourceParam());
+  }
+
+  protected override void SolveInstance(IGH_DataAccess da)
+  {
+    // get inputs
+    string urlInput = "";
+    if (!da.GetData(0, ref urlInput))
+    {
+      AddRuntimeMessage(GH_RuntimeMessageLevel.Warning, $"Speckle Url is missing");
+      return;
+    }
+
+    string tokenInput = "";
+    if (!da.GetData(1, ref tokenInput))
+    {
+      AddRuntimeMessage(GH_RuntimeMessageLevel.Warning, $"Speckle token is missing");
+      return;
+    }
+
+    try
+    {
+      // NOTE: once we split the logic in Sender and Receiver components, we need to set flag correctly
+      var (resource, hasPermission) = SolveInstanceWithUrAndToken(urlInput, tokenInput, true);
+      if (!hasPermission)
+      {
+        AddRuntimeMessage(GH_RuntimeMessageLevel.Error, "You do not have enough permission for this project.");
+      }
+      da.SetData(0, resource);
+    }
+    catch (SpeckleException e)
+    {
+      AddRuntimeMessage(GH_RuntimeMessageLevel.Error, e.Message);
+      da.AbortComponentSolution();
+    }
+  }
+
+  public (SpeckleUrlModelResource resource, bool hasPermission) SolveInstanceWithUrAndToken(
+    string input,
+    string token,
+    bool isSender
+  )
+  {
+    // When input is provided, lock interaction of buttons so only text is shown (no context menu)
+    // Should perform validation, fill in all internal data of the component (project, model, version, account)
+    // Should notify user if any of this goes wrong.
+
+    var resources = SpeckleResourceBuilder.FromUrlString(input, token);
+    if (resources.Length == 0)
+    {
+      throw new SpeckleException($"Input url string was empty");
+    }
+
+    if (resources.Length > 1)
+    {
+      throw new SpeckleException($"Input multi-model url is not supported");
+    }
+
+    var resource = resources.First();
+    using var scope = PriorityLoader.CreateScopeForActiveDocument();
+    var account = resource.Account.GetAccount(scope);
+    if (account != null)
+    {
+      scope.Get<IAccountService>().SetUserSelectedAccountId(account.id);
+    }
+    else
+    {
+      throw new SpeckleException("No account found for server URL");
```

Comment:
`GetAccount` can throw a few exceptions, would this block ever hit? we should inform the user if their token is expired or invalid - this doesn't seem to be the most descriptive message

Reply:
It's a new get account that basically always returns empty after checking fallbacks and whatnot.  Should not return a null ever?